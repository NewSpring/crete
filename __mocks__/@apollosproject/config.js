const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
    SHARE_URL: 'https://newspring.cc',
    TIMEZONE: 'America/New_York',
  },
  ROCK_CONSTANTS: {
    IMAGE: 10,
    AUDIO_FILE: 77,
    VIDEO_FILE: 79,
  },
  ROCK_MAPPINGS: {
    FEED_CONTENT_CHANNEL_IDS: [1, 2, 3, 4, 6, 8],
    SERIES_CONTENT_CHANNEL_TYPE_IDS: [6, 7],
    CONTENT_ITEM: {
      ContentSeriesContentItem: {
        EntityType: 'ContentChannelItem',
        ContentChannelId: [18, 20],
      },
      DevotionalContentItem: {
        EntityType: 'ContentChannelItem',
        ContentChannelId: [17],
      },
      MediaContentItem: {
        EntityType: 'ContentChannelItem',
      },
      UniversalContentItem: {
        EntityType: 'ContentChannelItem',
      },
      ContentItem: {
        EntityType: 'ContentChannelItem',
      },
      WeekendContentItem: {
        EntityType: 'ContentChannelItem',
      },
    },
  },
});

export default ApolloServer;
