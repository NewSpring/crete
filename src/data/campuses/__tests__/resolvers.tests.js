import { graphql } from 'graphql';
import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';
import { peopleSchema } from '@apollosproject/data-schema';
import * as Campus from '../index';

const { getSchema, getContext } = createTestHelpers({ Campus });

describe('Campus', () => {
  const context = getContext();
  context.dataSources.Campus.getPublicByLocation = jest.fn(() =>
    Promise.resolve([
      { location: { image: { path: 'https://fake-image-url.com' } } },
    ])
  );
  const schema = getSchema([peopleSchema]);
  it('gets campuses', async () => {
    const query = `
      query {
        campuses {
          image {
            uri
          }
        }
      }
    `;
    const result = await graphql(schema, query, {}, context);
    expect(result).toMatchSnapshot();
  });
});
