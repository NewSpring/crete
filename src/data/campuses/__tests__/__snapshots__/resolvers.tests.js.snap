// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campus gets campuses 1`] = `
Object {
  "data": Object {
    "campuses": Array [
      Object {
        "image": Object {
          "uri": "https://fake-image-url.com",
        },
      },
    ],
  },
}
`;
